config/config.go

```diff
@@ -88,46 +104,127 @@ var defaultConfig = &config{
 	},
 }
 
+// NewDefaultConfig returns a deep copy of the default configuration
+//
+// By doing this, we ensure that newConfig has its own independent copy of the MeteoProviders slice. Therefore, any changes made to newConfig will not affect defaultConfig.
+func NewDefaultConfig() *config {
+	newConfig := *defaultConfig
+	newConfig.MeteoProviders = make([]MeteoProvider, len(defaultConfig.MeteoProviders))
+	copy(newConfig.MeteoProviders, defaultConfig.MeteoProviders)
+	return &newConfig
+}
+
+// Get returns the current configuration that was last loaded, or loads the configuration if it hasn't been loaded yet
+func Get() (*config, error) {
+	if currentConfig == nil {
+		currentConfig, currentConfigErrors = Load(nil)
+		// return cfg, err
+	}
+	return currentConfig, currentConfigErrors
+}
 
-// New creates a new config object
-func New() (*config, error) {
-	conf := defaultConfig
+// Load loads the configuration from the file if a preferred config is not provided
+//
+// This is useful when Munch is being used as a package and the user wants to dynamically pass their own configuration
+func Load(c *config) (*config, error) {
+	currentConfig = NewDefaultConfig() // Assign it to config.currentConfig
 
-	pwd, err := os.Getwd()
-	cobra.CheckErr(err)
-	viper.AddConfigPath(pwd)
-	viper.AddConfigPath(".")
-	viper.SetConfigType("yml")
-	viper.SetConfigName("munch")
+	// Use the provided config if it's not nil
+	if c != nil {
+		currentConfig = c
+	}
 
-	if err := viper.ReadInConfig(); err != nil {
-		slog.Warn("Config file not found, using default config", "error", err)
-		return defaultConfig, nil 
+	// Load the config from the file if it hasn't been provided
+	if c == nil {
+		pwd, err := os.Getwd()
+		cobra.CheckErr(err)
+		viper.AddConfigPath(pwd)
+		viper.AddConfigPath(".")
+		viper.SetConfigType("yml")
+		viper.SetConfigName("munch")
+
+		if err := viper.ReadInConfig(); err != nil {
+			slog.Warn("Config file not found, using default config", "error", err)
+			return currentConfig, nil
+		}
+
+		if err := viper.Unmarshal(currentConfig); err != nil {
+			slog.Error(e.FAIL, "err", err, "description", "Couldn't parse config, using default config")
+			return currentConfig, nil
+		}
 	}
 
-	if err := viper.Unmarshal(conf); err != nil {
-		slog.Error(e.FAIL, "err", err, "description", "Couldn't parse config, using default config")
-		return defaultConfig, nil 
+	// Check for critical values, but do not panic. Instead, return an error
+	if err := validateCriticalFields(currentConfig, false); err != nil {
+		currentConfigErrors = err
+		return currentConfig, currentConfigErrors
 	}
-	// Check for critical values and panic if missing
-	validateCriticalFields(conf)
 
-	return conf, nil
+	currentConfigErrors = nil // Reset the errors
+	return currentConfig, currentConfigErrors
+}
+
+// CriticalError represents a custom error type for critical config errors
+type CriticalError struct {
+	Field   string
+	Message string
```

Comment:
I feel like renaming this to providerName or something would make it clearer (Ln 175,201,207,213), as Message leads us to believe that we are storing some error msg.

Reply:
Cool